modules/nw-dual-stack-convert.adoc

```diff
@@ -6,13 +6,28 @@ As a cluster administrator, you can convert your single-stack cluster network to
 
 Converting a single-stack cluster network to a dual-stack cluster network consists of creating patches and applying them to the cluster's network and infrastructure.
 
-If you need to add IPv6 Virtual IPs (VIPs) for API and Ingress services to an existing dual-stack-configured cluster, you need to patch only the cluster's infrastructure and not the cluster's network.
-
 [NOTE]
 ====
 Each patch operation that changes `clusterNetwork`, `serviceNetwork`, `apiServerInternalIPs`, and `ingressIP` objects triggers a restart of the cluster. Changing the `MachineNetworks` object does not cause a reboot of the cluster.
 ====
 
+If you need to add IPv6 virtual IPs (VIPs) for API and Ingress services to an existing dual-stack-configured cluster, you need to patch only the cluster's infrastructure and not the cluster's network.
+
+[IMPORTANT]
+====
+If you already upgraded your cluster to {product-title} 4.16 or later and you need to convert the single-stack cluster network to a dual-stack cluster network, you must specify an IPv4 network for API and Ingress services for your cluster in the YAML configuration patch file. This configuration ensures that IPv4 traffic exists in the same network interface as the default gateway.
```

Comment:
I guess it's the "specify an IPv4 network" that's confusing.  This is just the IPv4 `machineNetwork` from `install-config.yaml` right?

Admin has access to `install-config.yaml` so is there any reason admins shouldn't just use the `machineNetwork` from the `install-config.yaml`?

```
oc -n kube-system extract cm/cluster-config-v1 --to=- | sed -n  '/machineNetwork:/,/platform:/p'
```

or if `yq` is present.
 
```
oc -n kube-system extract cm/cluster-config-v1 --to=- | yq '.networking.machineNetwork[] | .cidr'
```

or check from the node routes themselves.

```
for f in $(oc get pod -n openshift-ovn-kubernetes -l app=ovnkube-node  --no-headers -o custom-columns=N:.metadata.name) ;  do oc -n openshift-ovn-kubernetes  exec -c ovn-controller $f -- ip -4 route show  | awk '/\/\d*.*br-ex.*metric/ {print $1}' ; done
```

But that's probably too complex.   This is okay for now I guess.  

Otherwise we could say: 

"you must specify the existing IPv4 machineNetwork for API and Ingress services for your cluster in the YAML configuration patch file."

Reply:
Depending on the organization, different people have different access and knowledge. We should stick to saying "use the same value you used in install-config" and let them figure out how to get it. The command here is almost a hack, something I'd never advise to use in a copy-pasteable doc. If they need help doing this they can always ask us, but I don't feel any of the 3 commands from the above should be written in a copy-pasteable form

Reply:
[Get install-config.yaml from an installed cluster RHOCP4 ](https://access.redhat.com/solutions/6963617) is published, so admins should be able to use it.

CI and e2e for day-2 single to dualstack plus VIPs is being implemented right now.  I'm currently testing the `ip -4 route show` method.   But either way we need an automated way to find the `machineNetwork` for CI testing.